packages/compass-aggregations/src/utils/cancellable-aggregation.ts

```diff
@@ -63,3 +63,54 @@ export async function aggregatePipeline({
 
   return result;
 }
+
+
+/**
+ * todo: move to data-service (COMPASS-5808)
+ */
+export async function explainPipeline({
+  dataService,
+  signal,
+  namespace,
+  pipeline,
+  options,
+}: {
+  dataService: DataService;
+  signal: AbortSignal;
+  namespace: string;
+  pipeline: Document[];
+  options: AggregateOptions;
+}): Promise<Document> {
+  if (signal.aborted) {
+    return Promise.reject(createCancelError());
+  }
+  const session = dataService.startSession('CRUD');
+  const cursor = dataService.aggregate(
+    namespace,
+    pipeline,
+    options
+  );
+  const abort = () => {
+    Promise.all([
+      cursor.close(),
+      dataService.killSessions(session)
+    ]).catch((err) => {
+      log.warn(
+        mongoLogId(1001000139),
```

Comment:
tiny nit, should be okay here since this is .ts:

```suggestion
        mongoLogId(1_001_000_139),
```